genesiscommunitysuccess/blank-app-seed
# fix: reduce apiHost steps when using genx in prompt mode - FUI-2207

[Review Guidance](https://www.notion.so/genesisglobal/Platform-Code-Review-Process-ace93ba760cc4563b0dfb712e2a88d8a)

📓 &nbsp; **Related JIRA**

[FUI-2207](https://genesisglobal.atlassian.net/browse/FUI-2207)

🤔 &nbsp; **What does this PR do?**

- Attempt on fixing "blank apiHost" after setting it in genx steps (non-prompt mode)

🚀 &nbsp; **Where should the reviewer start?**

See the `api.js` changes related to the `apiHost` question + how the value is set/used [here](https://github.com/genesiscommunitysuccess/blank-app-seed/pull/343/files#diff-f3b44cae9c48356e0cdcd428aecf336a498a1bc7eaca242d7a2598a5d75df2aeR14).

📑 &nbsp; **How should this be tested?**

First:

```
rm -rf blankappseedtest && npx -y @genesislcap/genx@latest init blankappseedtest --ref bw/api-host-fix
```

Then:

- set a custom API_HOST
- finish all the steps
- search for the value you used as API_HOST in the app, should be correctly set/ready to use

Repeat the above but without the `--ref` for this branch:

- Do the same, but on `main/prerelease` branch, "API_HOST" will remain the default... only works if setting in the actual genx command with `--apiHost` param.

✅ &nbsp; **Checklist**
